tracer/src/Datadog.Trace/DiagnosticListeners/DiagnosticManager.cs

```diff
@@ -3,23 +3,27 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
 // </copyright>
 
-#if !NETFRAMEWORK
+#nullable enable
+
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Reflection;
+using Datadog.Trace.DiagnosticListeners.DuckTypes;
+using Datadog.Trace.DuckTyping;
 using Datadog.Trace.Logging;
 using Datadog.Trace.Util;
 using Datadog.Trace.Vendors.Serilog.Events;
 
 namespace Datadog.Trace.DiagnosticListeners
 {
-    internal sealed class DiagnosticManager : IDiagnosticManager, IObserver<DiagnosticListener>, IDisposable
+    internal sealed class DiagnosticManager : IDiagnosticManager, IObserver<IDiagnosticListener>, IDisposable
```

Comment:
I have one refactor I'd like for you to apply: Using reflection you are now creating a type that implements `IObserver<DiagnosticListener>` so we no longer need to implement this interface. We should remove this to reduce confusion.

In addition to this code suggestion, you will also need to:
- Remove `void IObserver<IDiagnosticListener>.OnCompleted()`
- Remove `void IObserver<IDiagnosticListener>.OnError(Exception error)`
- Rename `void IObserver<IDiagnosticListener>.OnNext(IDiagnosticListener listener)` to `public void OnNext(IDiagnosticListener listener)`
- On line 181, change `((IObserver<IDiagnosticListener>)this).OnNext(listener);` to `OnNext(listener);`

```suggestion
    internal sealed class DiagnosticManager : IDiagnosticManager, IDisposable
```